src/main/java/io/logz/jmx2graphite/Jmx2GraphiteConfiguration.java

```diff
@@ -2,52 +2,82 @@
 
 import com.typesafe.config.Config;
 
+import java.net.InetAddress;
 import java.net.MalformedURLException;
 import java.net.URL;
+import java.net.UnknownHostException;
 import java.util.concurrent.TimeUnit;
 
 /**
  * @author amesika
  */
 public class Jmx2GraphiteConfiguration {
     /* Jolokia full URL */
-    public String jolokiaUrl;
+    private String jolokiaUrl;
 
-    public String graphiteHostname;
-
-    public int graphitePort;
+    // Graphite server related params
+    private  String graphiteHostname;
+    private int graphitePort;
 
     /* Short name of the sampled service (exposed by the Jolokia URL)", required = false */
-    public String serviceName = null;
+    private String serviceName = null;
 
     /* host of the sampled service (exposed by the Jolokia URL) */
-    public String serviceHost = null;
+    private String serviceHost = null;
 
     /* Metrics polling interval in seconds */
-    public int intervalInSeconds;
+    private int intervalInSeconds;
     private int graphiteConnectTimeout;
-    public int graphiteSocketTimeout;
+    private int graphiteSocketTimeout;
     private int graphiteWriteTimeoutMs;
 
-    public Jmx2GraphiteConfiguration(Config config) throws IllegalConfiguration {
-        jolokiaUrl = config.getString("service.jolokiaUrl");
-        String jolokiaHost;
-        try {
-            URL jolokia = new URL(jolokiaUrl);
-            jolokiaHost = jolokia.getHost();
-        } catch (MalformedURLException e) {
-            throw new IllegalConfiguration("service.jolokiaUrl must be a valid URL. Error = "+e.getMessage());
+    // Which client should we use
+    private MetricClientType metricClientType;
+
+    public enum MetricClientType {
+        JOLOKIA,
+        MBEAN_PLATFORM
+    }
+
+    public Jmx2GraphiteConfiguration(Config config, MetricClientType metricClientType) throws IllegalConfiguration {
+
+        this.metricClientType = metricClientType;
```

Comment:
I think it will be more elegant to embed this in the configuration, as we talked:
```
service {
  name = ${?SERVICE_NAME}
  host = ${?SERVICE_HOST}
  
  poller {
     // Option 1
     mbean-direct {
     }

     // OR

     jolokia {
          fullURL = 
     }
  }
 }
 
 graphite {
 hostname = ${?GRAPHITE_HOST}
 port = ${?GRAPHITE_PORT}
}

intervalInSeconds = ${?INTERVAL_IN_SEC} 
```